src/operator/controllers/intents_reconcilers/protected_services/should_protect.go

```diff
@@ -4,19 +4,27 @@ import (
 	"context"
 	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
 	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/samber/lo"
 	"github.com/sirupsen/logrus"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func IsServerEnforcementEnabledDueToProtectionOrDefaultState(ctx context.Context, kube client.Client, serverName string, serverNamespace string, enforcementDefaultState bool) (bool, error) {
+func IsServerEnforcementEnabledDueToProtectionOrDefaultState(ctx context.Context, kube client.Client, serverName string, serverNamespace string, enforcementDefaultState bool, activeNamespaces []string) (bool, error) {
 	if enforcementDefaultState {
 		logrus.Debug("Enforcement is default on, so all services should be protected")
 		return true, nil
 	}
+	logrus.Debug("Protected services are enabled")
 
-	logrus.Debug("Protected services are enabled, checking if server is in protected list")
+	logrus.Debugf("checking if server's namespace is in acrive namespaces")
+	if lo.Contains(activeNamespaces, serverNamespace) {
```

Comment:
Maybe turn the slice into a set on init? So This stays performant even with lots ofNamespaces